### ISSUE TITLE
Product cannot have overlay in vendor tree

### ISSUE THREAD

**Issue by NateDev100** [opengapps/aosp_build#146]
I'm trying to build AOSP for the Nexus 5X(bullhead) and using this aosp_build repo. Here's the error I get:

`build/core/tasks/vendor_module_check.mk:81: error: Error: Product "aosp_bullhead" cannot have overlay in vendor tree: vendor/opengapps/build/overlay/pico vendor/opengapps/build/overlay/webview/24 vendor/opengapps/build/overlay/browser vendor/opengapps/build/overlay/dialer vendor/opengapps/build/overlay/mms.
20:39:05 ckati failed with: exit status 1
build/core/main.mk:21: recipe for target 'run_soong_ui' failed
make: *** [run_soong_ui] Error 1

#### make failed to build some targets (02:54 (mm:ss)) ####
`

Is there a fix for this?

**Comment by Blystad** [comment]
Try setting:

```
PRODUCT_RESTRICT_VENDOR_FILES := false
```

In `BoardConfig.mk` and see if that helps. Or, in `device.mk`.

We moved stuff a bit around in #129 

@jamuir I think we'll see more and more devices using the vendor partition with Project Treble, so maybe we should add a fix for this. If setting it in `device.mk` will do, then we can add it to `opengapps-packages.mk`, otherwise we need something for `BoardConfig.mk`

**Comment by NateDev100** [comment]
Ok I've set that in BoardConfig.mk, will report back what happens, and then go and set it in device.mk, and see what happens. Thanks.

**Comment by jamuir** [comment]
@NateDev100:  we've hit that error in our in-house builds.

As Blystad mentioned, setting `PRODUCT_RESTRICT_VENDOR_FILES := false` will fix it.

Try changing it here:  `device/lge/bullhead/aosp_bullhead.mk`

**Comment by jamuir** [comment]
> @jamuir I think we'll see more and more devices using the vendor partition with Project Treble, so maybe we should add a fix for this. If setting it in device.mk will do, then we can add it to opengapps-packages.mk, otherwise we need something for BoardConfig.mk

@Blystad:  Agreed.  We are doing our Android O port now, so I will look for a good place to set `PRODUCT_RESTRICT_VENDOR_FILES := false`.

**Comment by NateDev100** [comment]
@jamuir Yep, I think that fixed it as it's gotten further along in the build than it did before. Glad it was solved so quickly. I assume that when Oreo Gapps are released, my repo sync will automatically pull in the Oreo builds? And I know I shouldn't really be asking this :wink: but is there an ETA for O?

**Comment by alviteri** [comment]
Sorry for the OT ... I have built Oreo and it works nice ..... Just two problems 

Google Play Store sometimes do not update packages until reboot device
Having BT on make device reboot all the time (a build whitout gapps has BT ok) I can't understand the cause 

![screenshot_20170901-223047](https://user-images.githubusercontent.com/5174751/29986785-3982cc6c-8f65-11e7-994f-0e708a4bc288.png)
![screenshot_20170901-223041](https://user-images.githubusercontent.com/5174751/29986786-398daf74-8f65-11e7-8a63-79dcdefc1942.png)
![screenshot_20170901-223027](https://user-images.githubusercontent.com/5174751/29986787-3992ac40-8f65-11e7-9aa6-f105c582d1fd.png)

**Comment by Blystad** [comment]
@NateDev100 Yes. On O: for any existing apps that come with O versions, we'll pick the O version automatically. If there's new GApps in O then we'll have to add Android makefiles for them. I welcome PRs to handle that :)

@alviteri Please open a separate issue, and share a full logcat.

**Comment by alviteri** [comment]
@Blystad on Oreo dialer needs sysconfig/dialer_experience.xml

<html>
<?xml version="1.0" encoding="utf-8"?>


<config>
    
    <feature name="com.google.android.apps.dialer.SUPPORTED" />
</config>
</html>

**Comment by mfonville** [comment]
See the config changes (in /etc/ ) of https://github.com/opengapps/all/commit/45c5900c54620fbe844240371accd73177df3898

**Comment by FeeJai** [comment]
Taimen devices also set `PRODUCT_RESTRICT_VENDOR_FILES := owner`. Is it the recommended way of installing OpenGAPPS to just modify the `aosp_taimen.mk`?

**Comment by jamuir** [comment]
@FeeJai :  yes, the recommended work-around is to set `PRODUCT_RESTRICT_VENDOR_FILES := false` in `aosp_taimen.mk`.

It may be possible to fix this by updating our `PRODUCT_COPY_FILES` statements.  We add entries of the form `<src>:<dest>`, but if we change them to `<src>:<dest>:<owner>`, then that may avoid the build error.

**Comment by jamuir** [comment]
> It may be possible to fix this by updating our `PRODUCT_COPY_FILES` statements. We add entries of the form `<src>:<dest>`, but if we change them to `<src>:<dest>:<owner>`, then that may avoid the build error.

It is more involved than this.

We also need an exemption that allows overlays to be picked up from inside `vendor/opengapps`.

Instructing users to set `PRODUCT_RESTRICT_VENDOR_FILES := false` in the aosp_build readme seems easiest.
